api/lanes.py

```diff
@@ -907,115 +917,100 @@ def _contributor_sublanes(self, _db):
 
     def _recommendation_sublane(self, _db, novelist_api):
         """Create a recommendations sublane."""
+        lane_name = "Recommendations for %s by %s" % (
+            self.work.title, self.work.author
+        )
         try:
-            lane_name = "Recommendations for %s by %s" % (
-                self.work.title, self.work.author
-            )
             recommendation_lane = RecommendationLane(
-                self.get_library(_db), self.work, lane_name, novelist_api=novelist_api,
+                library=self.get_library(_db), work=self.work,
+                display_name=lane_name, novelist_api=novelist_api,
                 parent=self,
             )
             if recommendation_lane.recommendations:
                 yield recommendation_lane
         except CannotLoadConfiguration, e:
-            # NoveList isn't configured.
+            # NoveList isn't configured. This isn't fatal -- we just won't
+            # use this sublane.
             pass
 
 
-class RecommendationLane(WorkBasedLane, WorksFromDatabase):
+class RecommendationLane(WorkBasedLane, DatabaseExclusiveWorkList):
     """A lane of recommended Works based on a particular Work"""
 
     DISPLAY_NAME = "Recommended Books"
     ROUTE = "recommendations"
     MAX_CACHE_AGE = 7*24*60*60      # one week
+    CACHED_FEED_TYPE = CachedFeed.RECOMMENDATIONS_TYPE
 
     def __init__(self, library, work, display_name=None,
                  novelist_api=None, parent=None):
+        """Constructor.
+
+        :raises: CannotLoadConfiguration if `novelist_api` is not provided
+        and no Novelist integration is configured for this library.
+        """
         super(RecommendationLane, self).__init__(
             library, work, display_name=display_name,
         )
-        _db = Session.object_session(library)
-        self.api = novelist_api or NoveListAPI.from_config(library)
-        self.recommendations = self.fetch_recommendations(_db)
+        self.novelist_api = novelist_api or NoveListAPI.from_config(library)
         if parent:
             parent.append_child(self)
+        _db = Session.object_session(library)
+        self.recommendations = self.fetch_recommendations(_db)
 
     def fetch_recommendations(self, _db):
         """Get identifiers of recommendations for this LicensePool"""
-
-        metadata = self.api.lookup(self.edition.primary_identifier)
+        metadata = self.novelist_api.lookup(self.edition.primary_identifier)
         if metadata:
             metadata.filter_recommendations(_db)
             return metadata.recommendations
         return []
 
-    def apply_filters(self, _db, qu, facets, pagination, featured=False):
```

Comment:
Unlike with most other lanes, this code wasn't removed -- it went into `modify_database_query_hook` below.